### PR TITLE
Only cache semantic classifications for C# and VB.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes [AB#1188251](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1188251)

Trying to find a good way to write a test here.

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/SemanticClassificationCache/SemanticClassificationCacheIncrementalAnalyzerProvider.cs [hunk 1]
```diff
@@ -40,6 +40,13 @@ private class SemanticClassificationCacheIncrementalAnalyzer : IncrementalAnalyz
         {
             public override async Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, InvocationReasons reasons, CancellationToken cancellationToken)
             {
+                // only process C# and VB.  OOP does not contain files for other languages.
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/SemanticClassificationCache/SemanticClassificationCacheIncrementalAnalyzerProvider.cs:43]: 💭 Seems like it _should_ contain those files

Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/SemanticClassificationCache/SemanticClassificationCacheIncrementalAnalyzerProvider.cs:43]: yup.  i've opened https://github.com/dotnet/roslyn/issues/47341 to track that.

**Comment by jinujoseph** [comment]
lets port this to 16.8.preview3

**Comment by CyrusNajmabadi** [comment]
@jinujoseph  have rebased.

**Review by davidwengier — APPROVED** [review]

**Pull request merged**